lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -714,6 +714,15 @@ bool ExportVerilog::isExpressionEmittedInline(Operation *op,
   if (!op->getResult(0).hasOneUse() && !isDuplicatableExpression(op))
     return false;
 
+  // Force array index expressions to be a simple name when the
+  // `disallowArrayIndexInlining` option is set.
+  if (options.disallowArrayIndexInlining && !isa<sv::ReadInOutOp>(op))
+    for (auto *user : op->getUsers())
```

Comment:
nit: there is already a loop over users to check a similar issue in `isExpressionUnableToInline`, so maybe this can go there? it's always felt a little weird having that divide, but this logic potentially fits in the "unable to inline" category.

Reply:
Good point! Let me move this there.

Reply:
Awesome, thanks.